compositor_pipeline/src/pipeline/output/rtp.rs

```diff
@@ -47,42 +63,142 @@ impl RtpSender {
+        Ok((socket, port))
+    }
+}
+
+fn run_tcp_sender_thread(
+    context: &mut RtpContext,
+    mut packets: Box<dyn Iterator<Item = EncodedChunk> + Send>,
+) {
+    // make accept non blocking so we have a chance to handle should_close value
+    context
+        .socket
+        .set_nonblocking(true)
+        .expect("Cannot set non-blocking");
+    loop {
+        if context
+            .should_close
+            .load(std::sync::atomic::Ordering::Relaxed)
+        {
+            return;
+        }
+
+        // accept only one connection at the time
+        let Ok((socket, _)) = context.socket.accept() else {
+            thread::sleep(Duration::from_millis(50));
+            continue;
+        };
+
+        let mut socket = TcpWritePacketStream::new(socket, context.should_close.clone());
+        loop {
+            let Some(EncodedChunk { data, pts, .. }) = packets.next() else {
+                return;
+            };
+
+            let payloads = match context.payloader.payload(64000, &data) {
+                Ok(p) => p,
+                Err(e) => {
+                    error!("Failed to payload a packet: {}", e);
+                    return;
+                }
+            };
+            let packets_amount = payloads.len();
+
+            for (i, payload) in payloads.into_iter().enumerate() {
+                let header = rtp::header::Header {
+                    version: 2,
+                    padding: false,
+                    extension: false,
+                    marker: i == packets_amount - 1, // marker needs to be set on the last packet of each frame
+                    payload_type: 96,
+                    sequence_number: context.next_sequence_number,
+                    timestamp: (pts.as_secs_f64() * 90000.0) as u32,
+                    ssrc: context.ssrc,
+                    ..Default::default()
+                };
+
+                let packet = rtp::packet::Packet { header, payload };
+
+                let packet = match packet.marshal() {
+                    Ok(p) => p,
+                    Err(e) => {
+                        error!("Failed to marshal a packet: {}", e);
+                        return;
+                    }
+                };
+
+                if let Err(err) = socket.write_packet(packet) {
+                    debug!("Failed to send packet: {err}");
+                }
+
+                context.next_sequence_number = context.next_sequence_number.wrapping_add(1);
+            }
+        }
+    }
+}
+
+/// this assumes, that a "packet" contains data about a single frame (access unit)
+fn run_udp_sender_thread(
+    context: &mut RtpContext,
+    packets: Box<dyn Iterator<Item = EncodedChunk> + Send>,
+) {
+    for packet in packets {
         // TODO: check if this is h264
         let EncodedChunk { data, pts, .. } = packet;
 
-        let payloads = match context.payloader.payload(1500, &data) {
+        let payloads = match context.payloader.payload(1400, &data) {
```

Comment:
MTU? I guess that padding is for headers etc., but you could explain where the magic value of 1400 comes from